### PR TITLE
Upgrade UIKit, Emotion, React

### PR DESCRIPTION
**Pull request by joneubank**
# Description of changes

Upgraded to React 17, Emotion 11, and UIKit 2 (which made the same version updates).
Removed dependency on arranger components - only one component (SQONView) was being used, due to version mismatch issues that component has been recreated in the platform-ui source.

## Type of Change

- [ ] Bug
- [x] Dependency updates
- [ ] Feature
- [ ] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [ ] Connected ticket to PR

### CODE REVIEW

File: pages/user/index.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import React from 'react';
+import * as React from 'react';
```
Comment (caravinci) [inline pages/user/index.tsx:20]: wut this?

File: pages/team.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 import TeamPage from 'components/pages/team';
 import { createPage } from 'global/utils/pages';
-import React from 'react';
+import * as React from 'react';
```
Comment (caravinci) [inline pages/team.tsx:22]: not bueno. 

React 17 doesn't even require this import

Comment (samrichca) [reply pages/team.tsx:22]: when I was working on it I had trouble with React imports so I added this

I'll look into webpack alias/resolve

File: pages/_app.tsx [hunk 3]
```diff
@@ -160,7 +160,8 @@ class Root extends App<
       throw err;
     }
 
-    const pageProps = await Component.getInitialProps({ ...ctx, egoJwt });
+    const pageProps =
+      (Component.getInitialProps && (await Component.getInitialProps({ ...ctx, egoJwt }))) || {};
```
Comment (caravinci) [inline pages/_app.tsx:164]: Thinking a ternary may make that more readable? I find the && is better used in short-circuits (where you don't expect a fallback)
```suggestion
      Component.getInitialProps
        ? await Component.getInitialProps({ ...ctx, egoJwt })
        : {};
```

File: pages/team.tsx (hunk 2 above)
Comment (joneubank) [reply pages/team.tsx:22]: not even a little bit sure when or why that got added. thanks for calling it out.

Comment (caravinci) [reply pages/team.tsx:22]: example from submission_ui
image

File: global/hooks/useProgramCheckEffect.tsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ export const useProgramCheckEffect = () => {
   const { setGlobalLoading, isGlobalLoading } = useGlobalLoader();
   const { isLoggingOut } = useAuthContext();
 
-  useEffect(() => {
+  React.useEffect(() => {
```
Comment (caravinci) [inline global/hooks/useProgramCheckEffect.tsx:41]: why not a destructured import instead, like with almost every other import in lines 20 - 39?

File: pages/team.tsx (hunk 2 above)
Comment (samrichca) [reply pages/team.tsx]: didn't end up being needed

File: Jenkinsfile.groovy [hunk 5]
```diff
@@ -69,14 +69,14 @@ spec:
...
         }
-        stage('Test') {
-            steps {
-                container('node') {
-                    sh "npm ci"
-                    sh "npm run test"
-                }
-            }
-        }
+        // stage('Test') {
```
Comment (demariadaniel) [inline Jenkinsfile.groovy:72]: Commented out? Not working in the pipeline?

File: components/SQONView/index.js [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+  //   emptyMessage: String,
+  //   sqon: TGroupSQON,
+  //   FieldCrumb: (props: TFieldCrumbArg) => any,
+  //   ValueCrumb: (props: TValueCrumbArg) => any,
+  //   Clear: (props: TClearArg) => any,
+  //   isExpanded: (valueSQON: TValueSQON) => boolean,
+  //   expanded: Array<TValueSQON>,
+  //   setExpanded: () => void,
+  //   onLessClicked: Function,
+  // }
```
Comment (demariadaniel) [inline components/SQONView/index.js:99]: Can these be commented back in?

File: components/pages/Homepage/common.tsx [hunk 7]
```diff
@@ -17,21 +17,26 @@
...
+  Typography,
+  UikitIconNames,
+  UikitTheme,
+  useTheme,
+} from '@icgc-argo/uikit';
 import { reactGridBreakpoints } from 'global/utils/common';
 import Image from 'next/image';
+import { ReactNode } from 'react';
+import { Col, Row } from 'react-grid-system';
+import { ComponentType } from 'react';
```
Comment (demariadaniel) [inline components/pages/Homepage/common.tsx:39]: lines 37 + 39 can be combined here right?
`import { ComponentType, ReactNode } from 'react';`

File: components/pages/contact/index.tsx [hunk 8]
```diff
@@ -17,52 +17,57 @@
...
+  FormHelperText,
+  Input,
+  InputLabel,
+  Link,
+  Select,
+  styled,
+  Textarea,
+  TOAST_VARIANTS,
+  Typography,
+  UikitTheme,
```
Comment (demariadaniel) [inline components/pages/contact/index.tsx:35]: Unused import `UikitTheme`

File: components/pages/file-repository/FacetPanel/SearchResultsMenu.tsx [hunk 9]
```diff
@@ -17,21 +17,16 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import React from 'react';
-import { css, styled } from '@icgc-argo/uikit';
-import Typography from '@icgc-argo/uikit/Typography';
-import Icon from '@icgc-argo/uikit/Icon';
+import { css, Icon, styled, Typography, UikitTheme, useTheme } from '@icgc-argo/uikit';
```
Comment (demariadaniel) [inline components/pages/file-repository/FacetPanel/SearchResultsMenu.tsx:20]: UikitTheme is unused here

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 10]
```diff
@@ -16,62 +16,69 @@
...
+  Icon,
+  Input,
+  MenuItem,
+  NumberRangeFacet,
+  styled,
+  SubMenu,
+  Tab,
+  Tabs,
+  Tooltip,
+  UikitTheme,
```
Comment (demariadaniel) [inline components/pages/file-repository/FacetPanel/index.tsx:34]: Here as well

Comment (demariadaniel) [reply components/pages/file-repository/FacetPanel/index.tsx:34]: This specific import is unused in at least 5 files, prefer if we clean it up before merging

File: Jenkinsfile.groovy (hunk 5 above)
Comment (joneubank) [reply Jenkinsfile.groovy:72]: yea we cant be doing this, we should figure out whats up with tests. thats my bad for skipping that locally before opening this.

File: components/pages/Homepage/common.tsx (hunk 7 above)
Comment (ciaranschutte) [reply components/pages/Homepage/common.tsx:39]: there's a vscode shortcut i use a bunch that will amalgamate imports & remove unused 
think it's this one: https://stackoverflow.com/questions/46722701/is-there-a-way-to-remove-unused-imports-and-declarations-from-angular-2

File: components/pages/file-repository/FacetPanel/index.tsx (hunk 10 above)
Comment (demariadaniel) [reply components/pages/file-repository/FacetPanel/index.tsx:34]: Updated

File: components/pages/Homepage/common.tsx (hunk 7 above)
Comment (demariadaniel) [reply components/pages/Homepage/common.tsx]: Updated

**Pull request merged**